game.py

```diff
@@ -43,22 +45,53 @@ def terminated(self) -> bool:
             return False
 
     def legal_actions(self) -> List[bool]:
-        if self.terminated():
-            return [False]*9
-        else:
-            return [x==0 for x in self.state]
+
+        moves = [False] * (64 * 64)
+
+        for move in self.board.legal_moves:
+            a = move.from_square
+            b = move.to_square
+            idx = (a * 64) + b
+            moves[idx] = True
+
+        return moves
     
     def next_state(self, action: int):
-        if self.state[action] == 0:
-            g = GameState(self)
-            g.state[action] = self.player()
-            return g
-        else:
+        legal_actions = self.legal_actions()
+        
+        if not legal_actions[action]:
             raise RuntimeError("Illegal action")
-    
+
+        g = GameState(self)
+        
+        a = action // 64
+        b = action % 64
+
+        move = chess.Move(a,b)
+        g.board.push(move)
+
+        return g     
+
+    "Returns representation of state suitable for input to neural network"
     def to_image(self):
-        "Returns representation of state suitable for input to neural network"
-        return np.array([self.state])
+
+        planes = []
+
+        for i in range(1,7):
+            plane = [  
+                
+                [
+                    (1 if ((a * 8) + b) in self.board.pieces(chess.PieceType(i), True) else 0)
```

Comment:
Bad code! @thegitone23 ???
Imagine what someone would think if they saw this on your github profile…